webkitgtk: honor systemdSupport

a-m-joseph opened the pull request:
This commit causes `webkitgtk` to honor the `systemdSupport` attr.

The default behavior (i.e. when this attrs is undefined) is unchanged.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

a-m-joseph commented:
> NOTE: I am currently running a rebuild of this. It may take a few hours.

Built sucessfully.

jtojnar commented on pkgs/development/libraries/webkitgtk/default.nix:
```diff
@@ -189,8 +195,10 @@ stdenv.mkDerivation rec {
+  ] ++ lib.optionals (!systemdSupport) [
     "-DUSE_SYSTEMD=OFF"
+  ] ++ lib.optionals (!dbusSupport) [
+    "-DDBUS_PROXY_EXECUTABLE=true"    # webkitgtk has a false bubblewrap->dbus dependency; this fixes it
```
I do not think this is a false dependency. The sandbox will [try to bind it](https://github.com/WebKit/WebKit/blob/2dbb11247d999671af1c8bd14837fcf4961e1c53/Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp#L682) and fail :

```ShellSession
$ nix shell -f ~/Projects/nixpkgs bubblewrap -c bwrap --ro-bind true true hello
bwrap: Can't find source path true: No such file or directory
```

And even if we replace it with something like `${coreutils}/bin/true`, that will not be on `PATH` in the sandbox, likely rendering webkit unusable.

a-m-joseph replied:
Thank you for the review.  I will need to investigate more closely to figure out how to explain to webkitgtk that if it is built without dbus support, it has no need for a dbus-proxy.  Apparently it does not know this.

In the meantime, I have revised this patch to deal only with systemdSupport.  I will submit dbusSupport as a separate patch.

Commit:
webkitgtk: honor systemdSupport

This commit exposes support for compilation without systemd,
controlled by the global systemdSupport argument.  This argument is
understood by many other nixpkgs expressions and can be set globally
in ~/.config/nixpkgs/config.nix.

a-m-joseph commented:
Latest push adds a commit message, no other changes.

Artturin commented on pkgs/development/libraries/webkitgtk/default.nix:
```diff
@@ -161,7 +162,9 @@ stdenv.mkDerivation rec {
     libseccomp
+  ] ++ lib.optionals systemdSupport [
     systemd
+  ] ++ lib.optionals stdenv.isLinux [
```
not needed, instead move the dependencies to the above isLinux

btw add your commit email to your github account so the commits get associated with your account

a-m-joseph replied:
Okay, done.  I was trying to minimize rebuilds by keeping the derivation the same in the default case, but you're right long-term maintainability matters more.

Artturin commented:
assuming you've tested building without systemdSupport

a-m-joseph commented:
Yes I have, on top of b7fe2da68bb9bb98d787eed6742772fca6fca367 from 2021-Dec-14.  I am currently trying to rebase to something more recent and will let you know in the unlikely event that the build fails.
